WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailFragment.kt

```diff
@@ -1,48 +1,185 @@
 package org.wordpress.android.ui.prefs.categories.detail
 
+import android.app.ProgressDialog
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
 import android.view.View
+import android.widget.AdapterView
+import androidx.annotation.StringRes
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
 import org.wordpress.android.R
+import org.wordpress.android.R.layout
 import org.wordpress.android.WordPress
 import org.wordpress.android.databinding.CategoryDetailFragmentBinding
-import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.models.CategoryNode
+import org.wordpress.android.ui.posts.ParentCategorySpinnerAdapter
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.Failure
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.InProgress
+import org.wordpress.android.ui.prefs.categories.detail.CategoryUpdateUiState.Success
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.util.ActivityUtils
+import org.wordpress.android.util.ToastUtils
+import org.wordpress.android.util.ToastUtils.Duration.SHORT
+import org.wordpress.android.viewmodel.observeEvent
 import javax.inject.Inject
 
+@Suppress("TooManyFunctions")
 class CategoryDetailFragment : Fragment(R.layout.category_detail_fragment) {
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     private lateinit var viewModel: CategoryDetailViewModel
+    @Inject lateinit var uiHelpers: UiHelpers
+    private lateinit var categoryAdapter: ParentCategorySpinnerAdapter
+
+    private var spinnerTouched: Boolean = false
+    private var mProgressDialog: ProgressDialog? = null
```

Comment:
🔍 
1. Hungarian notation can be removed from `mProgressDialog`.
2. Looks like `ProgressDialog` is deprecated. We might want to use a suggested alternative. However, I don't want to block this PR as I understand this logic was taken from an existing class. Feel free to create a separate task for it.

Reply:
@ashiagr Great catch on the Hungarian notation. Done in ff9101a. 

As for the `ProgressDialog`, I am planning to take it as a separate task and add an alternative element. This PR was already becoming huge so didn't want to burden the review with more changes. I will add it to the parent task. 👍

Reply:
I have added the task here https://github.com/wordpress-mobile/WordPress-Android/issues/16114. 👍